### DISCUSSION TITLE
Identifying skippable and escapable roles

### DISCUSSION THREAD

**Discussion by HadrienGardeur** [readium/guided-navigation#3] [General]
When using guided navigation, we want to make sure that users are free to automatically skip elements that are irrelevant to them or escape from a number of nested elements.

While escapability could be implemented as a general mechanism for nested elements, identifying specific roles might offer a better user experience.

EPUB 3.3 contains a small list of roles:

- [Skippable](https://www.w3.org/TR/epub-33/#sec-skippability)
  -  `endnote`
  - `footnote`
  - `pagebreak`
- [Escapable](https://www.w3.org/TR/epub-33/#sec-escapability)
  - `aside`
  - `figure`
  - `list`
  - `table`

Aside from a slightly longer list in older versions of the EPUB specification (referencing deprecated roles), I couldn't find a best practice document to cover skippability/escapability.

@danielweck are you aware of such a document? Could you document the list of roles currently supported in Thorium for that?

**Comment by HadrienGardeur** [comment]
Looking at DPUB ARIA 1.0 and EPUB 3 Structural Semantics Vocabulary, I see a number of roles that act as a container/collections, with an equivalent item element.

All of these roles are IMO good candidates for the list of escapable roles. 

Based on the current list of escapable roles, the following ones could be added:

- `list-item`
- `table-row`
- `table-cell`

Looking at DPUB ARIA, I could spot the following roles:

- `bibliography`
- `biblioentry`
- `credits`
- `credit`
- `endnotes`
- `endnote`
- `glossary` (child roles `glossterm` and `glossdef` are not included in DPUB ARIA)
- `index` (which has many child roles in EPUB 3 Structural Semantics Vocabulary, all missing from DPUB ARIA)

**Comment by HadrienGardeur** [comment]
Earlier this year, Vital Source shared some information about their approach to skippability: https://github.com/w3c/publishingcg/issues/72#issuecomment-1942724261

They offer two reading modes:
- simplified
- and detailed

The simplified mode automatically skips a number of roles that we've already identified, whereas the detailed one is much more comprehensive.

These roles are also grouped up in what they call filters:

- altText (`image` for us)
- figure (`figure`)
- table (`table`)
- citation (`footnotes`, `endnote` and `bibliography` for us)
- uriLink (we don't have an equivalent yet)
- and math (we're going to add `math` as well)

I'll need to test their app and see if these filters are configurable beyond the simplified/detailed modes, but overall it seems consistent with what we've documented so far.

**Comment by danielweck** [comment]
Hello Hadrien, there's a discussion about this in Thorium too: https://github.com/edrlab/thorium-reader/discussions/2655

**Reply by HadrienGardeur** [reply]
Thanks @danielweck, this is very useful.

To reply to @oscarlcarlsson, I'm tracking all the roles that we'd like to implement in Readium Web/Mobile through a registry on this repo: https://github.com/readium/guided-navigation/blob/main/roles.md

Since we're using a dedicated elements for descriptions (`description`) this could be a user setting across the board, rather than specific to a given role.

**Reply by danielweck** [reply]
Useful resources:
* https://www.w3.org/TR/epub-a11y-tech-11/#sync-skippability
* https://www.w3.org/TR/epub-a11y-tech-11/#sync-escapability

**Comment by gautierchomel** [comment]
Here are the links to the actual proposed [skippable roles](https://github.com/readium/guided-navigation/blob/main/roles.md#list-of-skippable-roles) and [escapable roles](https://github.com/readium/guided-navigation/blob/main/roles.md#list-of-escapable-roles)
